### ISSUE TITLE
Introduce a Ballerina connector for Zoom REST API

### ISSUE THREAD

**Issue by NipunaRanasinghe** [ballerina-platform/ballerina-library#7038]
We are planning to introduce a new Ballerina connector for the latest Zoom REST API, generated using it's OpenAPI specification.

Related links:
- https://marketplace.zoom.us/docs/api-reference/zoom-api
- https://developers.zoom.us/api-specs/zoom-api/methods/ZoomMeetingAPI-spec.json

This includes the following tasks:
1. Create a new repository under 'ballerina-platform' organization.
2. Review and sanitize the OpenAPI specification.
3. Generate the connector client.
4. Implement a test suite covering the core functionalities.
5. Write documentation with examples.

**Comment by s-vamshi** [comment]
hey @NipunaRanasinghe @keizer619 can i work on this issue?

**Comment by KasunUdaraWaththeHewa** [comment]
@NipunaRanasinghe @keizer619 please assign this issue to me :)

**Comment by MohamedSabthar** [comment]
👋 Welcome, @s-vamshi! 🚀

We're thrilled to have you join the [Ballerina Hacktoberfest](https://ballerina.io/hacktoberfest/) community! The issue has been assigned to you, and we’re excited to see your contributions.

To help you get started, here are a few essential resources:

1. **Connector Contributor Guide**: Make sure to go through our [Ballerina Hacktoberfest connector contributor guidelines](https://ballerina.io/hacktoberfest/connector-contributor-guide/) and follow the exact steps, to contribute effectively.
2. **Learn Ballerina**: Whether you're new to Ballerina or looking to enhance your skills, check out our [official learning resources](https://ballerina.io/learn/).
3. **Community & Support**: If you have technical questions, feel free to ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/ballerina) with the Ballerina tag, or join us on [Discord](https://discord.com/invite/ballerinalang) to connect with other community members.

No contribution is too small, and your feedback is always welcome! Don’t hesitate to ask questions, propose new ideas, or report issues.

We are currently in the process of creating a GitHub repository for this connector module and will update you once it’s available. In the meantime, please go through the relevant resources and documentation related to the connector.

**Happy coding** and welcome aboard! 🎉

**Comment by asmi55** [comment]
Hey could I work on this issue please?

**Comment by adibmbrk** [comment]
Hey, can i work on this issue?

**Comment by MohamedSabthar** [comment]
👋 Welcome, @adibmbrk! 🚀

We're thrilled to have you join the [Ballerina Hacktoberfest](https://ballerina.io/hacktoberfest/) community! The issue has been assigned to you, and we’re excited to see your contributions.

To help you get started, here are a few essential resources:

1. **Connector Contributor Guide**: Make sure to go through our [Ballerina Hacktoberfest connector contributor guidelines](https://ballerina.io/hacktoberfest/connector-contributor-guide/) and follow the exact steps, to contribute effectively.
2. **Learn Ballerina**: Whether you're new to Ballerina or looking to enhance your skills, check out our [official learning resources](https://ballerina.io/learn/).
3. **Community & Support**: If you have technical questions, feel free to ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/ballerina) with the Ballerina tag, or join us on [Discord](https://discord.com/invite/ballerinalang) to connect with other community members.

No contribution is too small, and your feedback is always welcome! Don’t hesitate to ask questions, propose new ideas, or report issues.

We are currently in the process of creating a GitHub repository for this connector module and will update you once it’s available. In the meantime, please go through the relevant resources and documentation related to the connector.

**Happy coding** and welcome aboard! 🎉

**Comment by adibmbrk** [comment]
I have generated the client succesfully, please let me know if any sanitation of the openapi spec is required. Link to the [repo](https://github.com/adibmbrk/module-ballerinax-zoom.zmeeting).

**Comment by TharmiganK** [comment]
Hi @adibmbrk ,

I'll be mentoring your project, and we have created the repository for your connector project with the initial project template: https://github.com/ballerina-platform/module-ballerinax-zoom.zmeeting.

**Please sync the changes in your repo with the `main` branch of the above repository and send a PR with the generated client. We will review the PR with let you know the next steps**

Please refer to the contribution process section for guidelines on raising a PR: https://ballerina.io/hacktoberfest/connector-contributor-guide/#contribution-process.

Happy coding! 🎉

**Comment by adibmbrk** [comment]
Hi @TharmiganK , I have generated the client successfully does it require any sanitations.

**Comment by TharmiganK** [comment]
Hi @adibmbrk, I will review the PR and let you know 👍🏼

**Comment by TharmiganK** [comment]
Hi @adibmbrk 
The given OpenAPI specification only has inline schemas which creates some random type names. Please regenerate the client using this [flatten OpenAPI specification](https://gist.github.com/TharmiganK/45e7bbb6ce3602affa97ba9881fbc36c)
and update the PR

**Comment by TharmiganK** [comment]
@adibmbrk I have added few comments in the PR, please check them

**Comment by TharmiganK** [comment]
@adibmbrk Are you still working on this issue?

**Comment by TharmiganK** [comment]
Hi @adibmbrk  👋

Just checking on the progress of the connector project.

Are you actively working on this project ? If not we will have to reassign this to someone else.

**Comment by thil4n** [comment]
Hi, I'd like to work on this.

**Comment by MohamedSabthar** [comment]
> Hi, I'd like to work on this.

Great! I’ve assigned the task to you—feel free to get started. Let us know if you need any support!

**Comment by TharmiganK** [comment]
Hi @thil4n, 👋 Welcome. You can start working on this by referring to this initial contribution by @adibmbrk : https://github.com/ballerina-platform/module-ballerinax-zoom.zmeeting/pull/2

**Comment by thil4n** [comment]
@TharmiganK I generated the client and added sanitization.  BTW, still not added the test cases and documentations.

**Comment by anupama-pathirage** [comment]
@thil4n Just checking in — we haven’t seen any updates on this issue for a while. To keep things moving, we’ll go ahead and remove the current assignment so that others in the community can take a look.

If you're still interested in working on this, feel free to comment here and we’ll be happy to reassign it to you. Thanks for your interest in contributing! 🙌

**Comment by manulthanura** [comment]
@anupama-pathirage is this still open for contribution?

**Comment by keizer619** [comment]
> [@anupama-pathirage](https://github.com/anupama-pathirage) is this still open for contribution?

Yes its open for contribution. If you are willing to contribute lets us know

**Comment by manulthanura** [comment]
Yes I'm willing!
@keizer619 Could you provide me with a reference for what I should do?

These are the things I've followed so far.
- [Contributing to Ballerina](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [Ballerina Zoom.zmeeting connector](https://github.com/ballerina-platform/module-ballerinax-zoom.zmeeting)

**Comment by ThisaruGuruge** [comment]
Closing the issue as we have two separate Zoom connectors:

* ballerinax/zoom.scheduler  
* ballerinax/zoom.meetings
